tests/components/evohome/test_storage.py

```diff
@@ -0,0 +1,199 @@
+
+
+class _SessionDataT(TypedDict):
+    sessionId: str
+
+
+class _TokenStoreT(TypedDict):
+    username: str
+    refresh_token: str
+    access_token: str
+    access_token_expires: str  # 2024-07-27T23:57:30+01:00
+    user_data: NotRequired[_SessionDataT]
+
+
+class _EmptyStoreT(TypedDict):
+    pass
+
+
+SZ_USERNAME: Final = "username"
+SZ_REFRESH_TOKEN: Final = "refresh_token"
+SZ_ACCESS_TOKEN: Final = "access_token"
+SZ_ACCESS_TOKEN_EXPIRES: Final = "access_token_expires"
+SZ_USER_DATA: Final = "user_data"
+
+
+def dt_pair(dt_dtm: datetime) -> tuple[datetime, str]:
+    """Return a datetime without milliseconds and its string representation."""
+    dt_str = dt_dtm.isoformat(timespec="seconds")  # e.g. 2024-07-28T00:57:29+01:00
+    return dt_util.parse_datetime(dt_str), dt_str  # type: ignore[return-value]
+
+
+ACCESS_TOKEN_EXP_DTM, ACCESS_TOKEN_EXP_STR = dt_pair(dt_util.now() + timedelta(hours=1))
+
+USERNAME_DIFF: Final = f"not_{USERNAME}"
+USERNAME_SAME: Final = USERNAME
+
+TEST_CONFIG: Final = {
+    CONF_USERNAME: USERNAME_SAME,
+    CONF_PASSWORD: "password",
+}
+
+TEST_DATA: Final[dict[str, _TokenStoreT]] = {
+    "sans_session_id": {
+        SZ_USERNAME: USERNAME_SAME,
+        SZ_REFRESH_TOKEN: REFRESH_TOKEN,
+        SZ_ACCESS_TOKEN: ACCESS_TOKEN,
+        SZ_ACCESS_TOKEN_EXPIRES: ACCESS_TOKEN_EXP_STR,
+    },
+    "with_session_id": {
+        SZ_USERNAME: USERNAME_SAME,
+        SZ_REFRESH_TOKEN: REFRESH_TOKEN,
+        SZ_ACCESS_TOKEN: ACCESS_TOKEN,
+        SZ_ACCESS_TOKEN_EXPIRES: ACCESS_TOKEN_EXP_STR,
+        SZ_USER_DATA: {"sessionId": SESSION_ID},
+    },
+}
+
+TEST_DATA_NULL: Final[dict[str, _EmptyStoreT | None]] = {
+    "store_is_absent": None,
+    "store_was_reset": {},
+}
+
+DOMAIN_STORAGE_BASE: Final = {
+    "version": STORAGE_VER,
+    "minor_version": 1,
+    "key": STORAGE_KEY,
+}
+
+
+@pytest.mark.parametrize("idx", TEST_DATA_NULL)
+async def test_auth_tokens_null(
+    hass: HomeAssistant,
+    hass_storage: dict[str, Any],
+    idx: str,
+) -> None:
+    """Test loading/saving authentication tokens when no cached tokens in the store."""
+
+    hass_storage[DOMAIN] = DOMAIN_STORAGE_BASE | {"data": TEST_DATA_NULL[idx]}
+
+    mock_client = await setup_evohome(hass, TEST_CONFIG)
+
+    # Confirm client was instantiated without tokens, as cache was empty...
+    assert SZ_REFRESH_TOKEN not in mock_client.call_args.kwargs
+    assert SZ_ACCESS_TOKEN not in mock_client.call_args.kwargs
+    assert SZ_ACCESS_TOKEN_EXPIRES not in mock_client.call_args.kwarg
+
+    # Confirm the expected tokens were cached to storage...
+    data: _TokenStoreT = hass_storage[DOMAIN]["data"]
+
+    assert data[SZ_USERNAME] == USERNAME_SAME
+    assert data[SZ_REFRESH_TOKEN] == f"new_{REFRESH_TOKEN}"
+    assert data[SZ_ACCESS_TOKEN] == f"new_{ACCESS_TOKEN}"
+    assert dt_util.parse_datetime(data[SZ_ACCESS_TOKEN_EXPIRES]) > dt_util.now()  # type: ignore[operator]
```

Comment:
We can pass `raise_on_error=True` to avoid the type error.
```suggestion
    assert dt_util.parse_datetime(data[SZ_ACCESS_TOKEN_EXPIRES], raise_on_error=True) > dt_util.now()
```